### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by cs-shubham-morankar**
PMDB Link: https://pmdb.fortinet.com/ProjectManagement/viewDocument.php?id=23212&doc_type=design

API Documentation: https://learn.microsoft.com/en-us/rest/api/intune/

UTCs:

- [x] Connector installation verified.
- [x] Connector image verified.
- [x] Actions and Playbooks list verified.
- [x] Check health verified. 
- [x] Playbook tags verified.

### CODE REVIEW

File: microsoft-intune/info.json [hunk 1]
```diff
@@ -0,0 +1,2295 @@
...
+      "output_schema": {},
+      "enabled": true
+    },
+    {
+      "title": "Get Location Details",
+      "description": "Retrieves specific information for a specific location for an hostname from Microsoft Intune.",
+      "operation": "get_location_by_hostname",
+      "category": "investigation",
+      "annotation": "get_location_by_hostname",
+      "parameters": [],
```
Comment (cs-prerna-joshi) [inline microsoft-intune/info.json:169]: hostname param is missing.

File: microsoft-intune/info.json [hunk 2]
```diff
@@ -0,0 +1,2295 @@
...
+          "type": "integer",
+          "name": "$top",
+          "tooltip": "(Optional) Specify the maximum number of results this operation should return, per page, in the response."
+        }
+      ],
+      "output_schema": {},
+      "enabled": true
+    },
+    {
+      "title": "Get IOS Policy Details",
```
Comment (cs-prerna-joshi) [inline microsoft-intune/info.json]:
```suggestion
      "title": "Get iOS Policy Details",
```

File: microsoft-intune/info.json [hunk 3]
```diff
@@ -0,0 +1,2295 @@
...
+          "description": "(Optional) Specify the select query to retrieve specific fields from Microsoft Intune. Refer link for more info: https://learn.microsoft.com/en-us/odata/concepts/queryoptions-overview#select",
+          "tooltip": "(Optional) Specify the select query to retrieve specific fields from Microsoft Intune. Refer link for more info: https://learn.microsoft.com/en-us/odata/concepts/queryoptions-overview#select",
+          "placeholder": "e.g. id, location, name, etc."
+        }
+      ],
+      "output_schema": {},
+      "enabled": true
+    },
+    {
+      "title": "Update IOS Policy",
```
Comment (cs-prerna-joshi) [inline microsoft-intune/info.json]:
```suggestion
      "title": "Update iOS Policy",
```

File: microsoft-intune/operations.py [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+def wipe_device_for_user(config, params, connector_info):
+    url = "/providers/Microsoft.Intune/locations/{0}/users/{1}/devices/{2}/wipe?api-version=2015-01-14-preview".format(
+        params.get('hostName'), params.get('userName'), params.get('deviceName'))
+    response = api_request("POST", url, connector_info, config)
+    return response
+
+
+def _check_health(config, connector_info):
+    try:
+        return check(config, connector_info)
```
Comment (cs-prerna-joshi) [inline microsoft-intune/operations.py]: After generating the token. please also check the one of the endpoint is accessible or not.

File: microsoft-intune/info.json (hunk 1 above)
Comment (cs-shubham-morankar) [reply microsoft-intune/info.json:169]: Not Required for this operation.
Jfyi: https://learn.microsoft.com/en-us/rest/api/intune/get-locations/get-locations

File: microsoft-intune/operations.py (hunk 4 above)
Comment (cs-shubham-morankar) [reply microsoft-intune/operations.py]: Not accessible.

Description: After executing actions message throws an error "Resource not found Microsoft.Intune" same error occurs when we execute from Microsoft REST API: https://learn.microsoft.com/en-us/rest/api/intune/get-locations/get-locations#code-try-0

File: microsoft-intune/microsoft_api_auth.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+from requests import request
+from time import time, ctime
+from datetime import datetime
+from connectors.core.connector import get_logger, ConnectorError
+
+CONFIG_SUPPORTS_TOKEN = True
```
Comment (cs-suyog-jinturkar) [inline microsoft-intune/microsoft_api_auth.py]: All imports at one place and all consts should be at one place for better code readability

File: microsoft-intune/operations.py [hunk 6]
```diff
@@ -0,0 +1,574 @@
+""" Copyright start
+  Copyright (C) 2008 - 2022 Fortinet Inc.
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+from connectors.core.connector import get_logger, ConnectorError
+from requests import exceptions as req_exceptions
```
Comment (cs-suyog-jinturkar) [inline microsoft-intune/operations.py:8]: Seems we haven't used pycharm plugin here? any idea why?

File: microsoft-intune/operations.py [hunk 7]
```diff
@@ -0,0 +1,574 @@
...
+    url = "/providers/Microsoft.Intune/locations/{0}/iosPolicies/{1}/apps/{2}?api-version=2015-01-14-preview".format(
+        params.get('hostName'), params.get('policyName'), params.get('appName'))
+    response = api_request("DELETE", url, connector_info, config)
+    return response
+
+
+def add_group_to_ios_policy(config, params, connector_info):
+    url = "/providers/Microsoft.Intune/locations/{0}/iosPolicies/{1}/groups/{2}?api-version=2015-01-14-preview".format(
+        params.get('hostName'), params.get('policyName'), params.get('groupId'))
+    response = api_request("POST", url, connector_info, config)
```
Comment (cs-suyog-jinturkar) [inline microsoft-intune/operations.py:509]: Good coding practice: use constants for hardcoded string if its used at multiple places, i.e. POST, GET, etc.

Comment (cs-suyog-jinturkar) [reply microsoft-intune/operations.py:509]: same with diff params

**Pull request merged**